Marsiple/Common/Network/MartianApiManager.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  DataFetcher.swift
+//  Marsiple
+//
+//  Created by Matej Korman on 23/07/2018.
+//  Copyright © 2018 Matej Korman. All rights reserved.
+//
+
+import Foundation
+
+class MartianApiManager {
+    static func getPosts(success: @escaping ([Post]) -> Void, failure: @escaping (LocalizedError) -> Void) {
+        guard let url = MartianApi.url?.appendingPathComponent(MartianApi.URLs.posts)
+            else { return DispatchQueue.main.async { failure(DataFetcherError.urlCreationFailure) } }
+        getData(url: url, success: success, failure: failure)
+    }
+    
+    static func getComments(forPostId id: Int, success: @escaping ([Comment]) -> Void, failure: @escaping (LocalizedError) -> Void) {
+        guard let url = MartianApi.url?
+                            .appendingPathComponent(MartianApi.URLs.posts)
+                            .appendingPathComponent("\(id)")
+                            .appendingPathComponent(MartianApi.URLs.comments)
+            else { return DispatchQueue.main.async { failure(DataFetcherError.urlCreationFailure) } }
+        getData(url: url, success: success, failure: failure)
+    }
+    
+    static func postComment(comment: Comment) {
+        guard let url = MartianApi.url?
+                            .appendingPathComponent(MartianApi.URLs.posts)
+                            .appendingPathComponent("\(comment.postId)")
+                            .appendingPathComponent(MartianApi.URLs.comments)
+            else { return }
+        postData(data: comment, url: url)
+    }
+}
+
+private extension MartianApiManager {
+    static func getData<DataType: Codable>(url: URL, success: @escaping ([DataType]) -> Void, failure: @escaping (LocalizedError) -> Void) {
+        let request = setupRequest(url: url, method: HTTPRequestMethod.get)
+        let session = URLSession.shared
+        let task = session.dataTask(with: request) { data, response, error in
+            if let error = error {
+                return DispatchQueue.main.async { failure(DataFetcherError.generic(error)) }
+            }
+            do {
+                guard let unwrappedData = data
+                    else { return DispatchQueue.main.async { failure(DataFetcherError.dataUnwrapingFailure) } }
+                let posts = try JSONDecoder().decode([DataType].self, from: unwrappedData)
+                DispatchQueue.main.async { success(posts) }
+            } catch {
+                DispatchQueue.main.async { failure(DataFetcherError.parsingDataFailure) }
+            }
+        }
+        task.resume()
+    }
+    
+    static func postData<DataType: Codable>(data: DataType, url: URL) {
```

Comment:
Posting needs success and failure closures. We already talked about this one in another PR, is this resolved somewhere?